internal/provider/provider.go

```diff
@@ -104,14 +106,30 @@ func (p *IncidentProvider) Configure(ctx context.Context, req provider.Configure
 		panic(bearerTokenProviderErr)
 	}
 
-	base := cleanhttp.DefaultClient()
-	base.Transport = &loghttp.Transport{
+	base := retryablehttp.NewClient()
+	base.RetryMax = 10
+	base.Backoff = func(min, max time.Duration, attemptNum int, httpResp *http.Response) time.Duration {
+		// Retry for rate limits and server errors.
+		if httpResp != nil && httpResp.StatusCode == http.StatusTooManyRequests {
+			retryAfter := httpResp.Header.Get("Retry-After")
+			if retryAfter != "" {
+				if retryAfterSeconds, err := time.ParseDuration(retryAfter + "s"); err == nil {
```

Comment:
We return an RFC1123 timestamp rather than a duration, so this won't parse the `Retry-After` header correctly. I've got a WIP PR here that parses the header: https://github.com/incident-io/catalog-importer/pull/138